src/ui/public/vis/agg_config.js

```diff
@@ -26,334 +26,332 @@
 
 import _ from 'lodash';
 import { fieldFormats } from '../registry/field_formats';
-import { aggTypes } from '../agg_types/index';
 
-export function AggConfig(vis, opts) {
-  const self = this;
+class AggConfig {
+
+  /**
+   * Ensure that all of the objects in the list have ids, the objects
+   * and list are modified by reference.
+   *
+   * @param  {array[object]} list - a list of objects, objects can be anything really
+   * @return {array} - the list that was passed in
+   */
+  static ensureIds(list) {
+    const have = [];
+    const haveNot = [];
+    list.forEach(function (obj) {
+      (obj.id ? have : haveNot).push(obj);
+    });
 
-  self.id = String(opts.id || AggConfig.nextId(vis.aggs));
-  self.vis = vis;
-  self._opts = opts = (opts || {});
-  self.enabled = typeof opts.enabled === 'boolean' ? opts.enabled : true;
+    let nextId = AggConfig.nextId(have);
+    haveNot.forEach(function (obj) {
+      obj.id = String(nextId++);
+    });
 
-  // start with empty params so that checks in type/schema setters don't freak
-  // because self.params is undefined
-  self.params = {};
+    return list;
+  }
 
-  // setters
-  self.type = opts.type;
-  self.schema = opts.schema;
+  /**
+   * Calculate the next id based on the ids in this list
+   *
+   * @return {array} list - a list of objects with id properties
+   */
+  static nextId(list) {
+    return 1 + list.reduce(function (max, obj) {
+      return Math.max(max, +obj.id || 0);
+    }, 0);
+  }
 
-  // set the params to the values from opts, or just to the defaults
-  self.setParams(opts.params || {});
-}
+  constructor(vis, opts) {
+    this.id = String(opts.id || AggConfig.nextId(vis.aggs));
+    this.vis = vis;
+    this._opts = opts = (opts || {});
+    this.enabled = typeof opts.enabled === 'boolean' ? opts.enabled : true;
 
-AggConfig.aggTypes = aggTypes;
```

Comment:
I removed this completly, since it was only used in one test.. and it wasn't even used there at all. Also this was causing an issue with cyclic dependencies, so I rather removed it.